### DISCUSSION TITLE
Help spinning up Docker server image to be used internally and externally

### DISCUSSION THREAD

**Discussion by itemforty** [Lurkars/gloomhavensecretariat#256] [Q&A] [answered]
I'm new to Docker and running an externally facing server in general, but I think I'm pretty close to getting this running. A few questions should sort me out:

1. I'm forwarding port 58888 , what should the docker compose "ports" mapping be? [localhost]:58888:8080?
2. Do I need to add inbound and outbound Firewall rules, or will Docker handle that?
3. When playing on the same network, what do I put in GHS client when I'm at home?
4. After forwarding the ports through the router, what do I put in the client when I'm not? I would think it's the IP the ISP assigns us, but I don't think that's static? Am I over thinking that?
5. When GHS is updated, is it as simple as reloading the container and updating the clients?

Sorry if any of these questions are dumb, and thanks for any help!

**Comment by Lurkars** [comment, marked as answer]
Also no expert 😅 but here my answers
> 1. I'm forwarding port 58888 , what should the docker compose "ports" mapping be? [localhost]:58888:8080?

Yes, that should be correct syntax is [external port]:[internal port]

> 2. Do I need to add inbound and outbound Firewall rules, or will Docker handle that?

This depends on your setup, if the socket host machine is blocking all (or at least your 58888) ports, you need to add firewall rules of course. Would be really dangerous if a docker container would handle host firewall.

> 3. When playing on the same network, what do I put in GHS client when I'm at home?

Local IP of the server, mostly something starting with `192.168...`

> 4. After forwarding the ports through the router, what do I put in the client when I'm not? I would think it's the IP the ISP assigns us, but I don't think that's static? Am I over thinking that?

Yes, need port forwarding on router and then your public IP address from ISP. You can also try to setup DynDNS and then use that host.

> 5. When GHS is updated, is it as simple as reloading the container and updating the clients?

I am unsure about the specific setup. Possible you need to call docker pull or sth. similar to update server, but in general should already update on restart! If you serving client with auto-update client setting, restart of server is enough for getting latest client version!

**Comment by itemforty** [comment]
Very helpful, finally got something running online. Thanks much!
